[Concurrent Segment Search][BUG] org.opensearch.search.profile.query.QueryProfilerIT.testProfileQuery {p0={"search.concurrent_segment_search.enabled":"true"} p1=true} is flaky

**Describe the bug**

The test case org.opensearch.search.profile.query.QueryProfilerIT.testProfileQuery {p0={"search.concurrent_segment_search.enabled":"true"} p1=true} is flaky:

```
Failed to execute phase [query], all shards failed; shardFailures {[NozaKKq8RO29Gdqj-Bi9WQ][test][0]: RemoteTransportException[[node_s4][127.0.0.1:41257][indices:data/read/search[phase/query]]]; nested: NullPointerException[Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null]; }{[jmngMMEcTbqleNF9QtB2ZA][test][1]: RemoteTransportException[[node_s3][127.0.0.1:37065][indices:data/read/search[phase/query]]]; nested: QueryPhaseExecutionException[Query Failed [Failed to execute main query]]; nested: QueryPhaseExecutionException[Query Failed [Failed to execute concurrent segment search thread]]; nested: NotSerializableExceptionWrapper[assertion_error: null]; }{[NozaKKq8RO29Gdqj-Bi9WQ][test][2]: RemoteTransportException[[node_s4][127.0.0.1:41257][indices:data/read/search[phase/query]]]; nested: NullPointerException[Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null]; }
```

```
Failed to execute phase [query], all shards failed; shardFailures {[NozaKKq8RO29Gdqj-Bi9WQ][test][0]: RemoteTransportException[[node_s4][127.0.0.1:41257][indices:data/read/search[phase/query]]]; nested: NullPointerException[Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null]; }{[jmngMMEcTbqleNF9QtB2ZA][test][1]: RemoteTransportException[[node_s3][127.0.0.1:37065][indices:data/read/search[phase/query]]]; nested: QueryPhaseExecutionException[Query Failed [Failed to execute main query]]; nested: QueryPhaseExecutionException[Query Failed [Failed to execute concurrent segment search thread]]; nested: NotSerializableExceptionWrapper[assertion_error: null]; }{[NozaKKq8RO29Gdqj-Bi9WQ][test][2]: RemoteTransportException[[node_s4][127.0.0.1:41257][indices:data/read/search[phase/query]]]; nested: NullPointerException[Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null]; }
	at __randomizedtesting.SeedInfo.seed([E0A359380AF3E50D:1B83BAD4F773DAEC]:0)
	at app//org.opensearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:677)
	at app//org.opensearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:373)
	at app//org.opensearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:716)
	at app//org.opensearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:485)
	at app//org.opensearch.action.search.AbstractSearchAsyncAction$1.onFailure(AbstractSearchAsyncAction.java:295)
	at app//org.opensearch.action.search.SearchExecutionStatsCollector.onFailure(SearchExecutionStatsCollector.java:104)
	at app//org.opensearch.action.ActionListenerResponseHandler.handleException(ActionListenerResponseHandler.java:75)
	at app//org.opensearch.action.search.SearchTransportService$ConnectionCountingHandler.handleException(SearchTransportService.java:749)
	at app//org.opensearch.transport.TransportService$6.handleException(TransportService.java:875)
	at app//org.opensearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1483)
	at app//org.opensearch.transport.InboundHandler.lambda$handleException$3(InboundHandler.java:421)
	at app//org.opensearch.common.util.concurrent.OpenSearchExecutors$DirectExecutorService.execute(OpenSearchExecutors.java:341)
	at app//org.opensearch.transport.InboundHandler.handleException(InboundHandler.java:419)
	at app//org.opensearch.transport.InboundHandler.handlerResponseError(InboundHandler.java:411)
	at app//org.opensearch.transport.InboundHandler.messageReceived(InboundHandler.java:159)
	at app//org.opensearch.transport.InboundHandler.inboundMessage(InboundHandler.java:115)
	at app//org.opensearch.transport.TcpTransport.inboundMessage(TcpTransport.java:767)
	at app//org.opensearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:175)
	at app//org.opensearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:150)
	at app//org.opensearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:115)
	at app//org.opensearch.transport.nio.MockNioTransport$MockTcpReadWriteHandler.consumeReads(MockNioTransport.java:341)
	at app//org.opensearch.nio.SocketChannelContext.handleReadBytes(SocketChannelContext.java:246)
	at app//org.opensearch.nio.BytesChannelContext.read(BytesChannelContext.java:59)
	at app//org.opensearch.nio.EventHandler.handleRead(EventHandler.java:152)
	at app//org.opensearch.transport.nio.TestEventHandler.handleRead(TestEventHandler.java:167)
	at app//org.opensearch.nio.NioSelector.handleRead(NioSelector.java:438)
	at app//org.opensearch.nio.NioSelector.processKey(NioSelector.java:264)
	at app//org.opensearch.nio.NioSelector.singleLoop(NioSelector.java:191)
	at app//org.opensearch.nio.NioSelector.runLoop(NioSelector.java:148)
	at java.base@20.0.2/java.lang.Thread.run(Thread.java:1623)
Caused by: OpenSearchException[Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null]; nested: NullPointerException[Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null];
	at app//org.opensearch.OpenSearchException.guessRootCauses(OpenSearchException.java:708)
	at app//org.opensearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:371)
	... 28 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "java.util.Map.get(Object)" is null
	at org.opensearch.search.profile.query.ConcurrentQueryProfileBreakdown.buildSliceLevelBreakdown(ConcurrentQueryProfileBreakdown.java:185)
	at org.opensearch.search.profile.query.ConcurrentQueryProfileBreakdown.toBreakdownMap(ConcurrentQueryProfileBreakdown.java:83)
	at org.opensearch.search.profile.query.ConcurrentQueryProfileTree.createProfileResult(ConcurrentQueryProfileTree.java:43)
	at org.opensearch.search.profile.query.ConcurrentQueryProfileTree.createProfileResult(ConcurrentQueryProfileTree.java:24)
	at org.opensearch.search.profile.AbstractInternalProfileTree.doGetTree(AbstractInternalProfileTree.java:183)
	at org.opensearch.search.profile.AbstractInternalProfileTree.doGetTree(AbstractInternalProfileTree.java:174)
	at org.opensearch.search.profile.AbstractInternalProfileTree.doGetTree(AbstractInternalProfileTree.java:174)
	at org.opensearch.search.profile.AbstractInternalProfileTree.getTree(AbstractInternalProfileTree.java:155)
	at org.opensearch.search.profile.query.ConcurrentQueryProfileTree.getTree(ConcurrentQueryProfileTree.java:73)
	at org.opensearch.search.profile.AbstractProfiler.getTree(AbstractProfiler.java:69)
	at org.opensearch.search.profile.SearchProfileShardResults.buildShardResults(SearchProfileShardResults.java:202)
	at org.opensearch.search.query.QueryPhase.execute(QueryPhase.java:162)
	at org.opensearch.search.SearchService.loadOrExecuteQueryPhase(SearchService.java:546)
	at org.opensearch.search.SearchService.executeQueryPhase(SearchService.java:610)
	at org.opensearch.search.SearchService$2.lambda$onResponse$0(SearchService.java:579)
	at org.opensearch.action.ActionRunnable.lambda$supply$0(ActionRunnable.java:74)
	at org.opensearch.action.ActionRunnable$2.doRun(ActionRunnable.java:89)
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
	at org.opensearch.threadpool.TaskAwareRunnable.doRun(TaskAwareRunnable.java:78)
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
	at org.opensearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:59)
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:908)
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.lang.Thread.run(Thread.java:1623)
```

**To Reproduce**

```
./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.profile.query.QueryProfilerIT" -Dtests.method="testProfileMatchesRegular {p0={"search.concurrent_segment_search.enabled":"true"} p1=true}" -Dtests.seed=E0A359380AF3E50D 
```


**Expected behavior**
A clear and concise description of what you expected to happen.

**Plugins**
Standard

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Host/Environment (please complete the following information):**
 - CI

**Additional context**
[Add any other context about the problem here.](https://build.ci.opensearch.org/job/gradle-check/24535/testReport/junit/org.opensearch.search.profile.query/QueryProfilerIT/testProfileQuery__p0___search_concurrent_segment_search_enabled___true___p1_true_/)

@ticheng-aws this is another one, but this time it is NPE, please take a look

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -187,6 +187,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Cleanup Unreferenced file on segment merge failure ([#9503](https://github.com/opensearch-project/OpenSearch/pull/9503))
 - Move ZStd to a plugin ([#9658](https://github.com/opensearch-project/OpenSearch/pull/9658))
 - [Remote Store] Add support for Remote Translog Store upload stats in `_nodes/stats/` API ([#8908](https://github.com/opensearch-project/OpenSearch/pull/8908))
+- Mute the query profile IT with concurrent execution ([#9840](https://github.com/opensearch-project/OpenSearch/pull/9840))
 
 ### Deprecated
 

--- a/server/src/internalClusterTest/java/org/opensearch/search/profile/query/QueryProfilerIT.java
+++ b/server/src/internalClusterTest/java/org/opensearch/search/profile/query/QueryProfilerIT.java
@@ -32,8 +32,6 @@
 
 package org.opensearch.search.profile.query;
 
-import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
-
 import org.apache.lucene.tests.util.English;
 import org.opensearch.action.index.IndexRequestBuilder;
 import org.opensearch.action.search.MultiSearchResponse;
@@ -42,23 +40,20 @@ import org.opensearch.action.search.SearchResponse;
 import org.opensearch.action.search.SearchType;
 import org.opensearch.action.search.ShardSearchFailure;
 import org.opensearch.common.settings.Settings;
-import org.opensearch.common.util.FeatureFlags;
 import org.opensearch.index.query.QueryBuilder;
 import org.opensearch.index.query.QueryBuilders;
 import org.opensearch.search.SearchHit;
 import org.opensearch.search.profile.ProfileResult;
 import org.opensearch.search.profile.ProfileShardResult;
 import org.opensearch.search.sort.SortOrder;
-import org.opensearch.test.ParameterizedOpenSearchIntegTestCase;
+import org.opensearch.test.OpenSearchIntegTestCase;
 
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import static org.opensearch.search.SearchService.CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING;
 import static org.opensearch.search.profile.query.RandomQueryGenerator.randomQueryBuilder;
 import static org.hamcrest.Matchers.emptyOrNullString;
 import static org.hamcrest.Matchers.equalTo;
@@ -66,32 +61,8 @@ import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
-import static org.hamcrest.Matchers.nullValue;
 
-public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
-    private final boolean concurrentSearchEnabled;
-    private static final String MAX_PREFIX = "max_";
-    private static final String MIN_PREFIX = "min_";
-    private static final String AVG_PREFIX = "avg_";
-    private static final String TIMING_TYPE_COUNT_SUFFIX = "_count";
-
-    public QueryProfilerIT(Settings settings, boolean concurrentSearchEnabled) {
-        super(settings);
-        this.concurrentSearchEnabled = concurrentSearchEnabled;
-    }
-
-    @ParametersFactory
-    public static Collection<Object[]> parameters() {
-        return Arrays.asList(
-            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), false).build(), false },
-            new Object[] { Settings.builder().put(CLUSTER_CONCURRENT_SEGMENT_SEARCH_SETTING.getKey(), true).build(), true }
-        );
-    }
-
-    @Override
-    protected Settings featureFlagSettings() {
-        return Settings.builder().put(super.featureFlagSettings()).put(FeatureFlags.CONCURRENT_SEGMENT_SEARCH, "true").build();
-    }
+public class QueryProfilerIT extends OpenSearchIntegTestCase {
 
     /**
      * This test simply checks to make sure nothing crashes.  Test indexes 100-150 documents,
@@ -258,7 +229,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertEquals(result.getLuceneDescription(), "field1:one");
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -301,7 +271,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
                     assertEquals(result.getProfiledChildren().size(), 2);
-                    assertQueryProfileResult(result);
 
                     // Check the children
                     List<ProfileResult> children = result.getProfiledChildren();
@@ -313,14 +282,12 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertThat(childProfile.getTime(), greaterThan(0L));
                     assertNotNull(childProfile.getTimeBreakdown());
                     assertEquals(childProfile.getProfiledChildren().size(), 0);
-                    assertQueryProfileResult(childProfile);
 
                     childProfile = children.get(1);
                     assertEquals(childProfile.getQueryName(), "TermQuery");
                     assertEquals(childProfile.getLuceneDescription(), "field1:two");
                     assertThat(childProfile.getTime(), greaterThan(0L));
                     assertNotNull(childProfile.getTimeBreakdown());
-                    assertQueryProfileResult(childProfile);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -363,7 +330,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertNotNull(result.getLuceneDescription());
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -409,7 +375,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertNotNull(result.getLuceneDescription());
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -450,7 +415,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertNotNull(result.getLuceneDescription());
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -491,7 +455,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertNotNull(result.getLuceneDescription());
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -531,7 +494,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertNotNull(result.getLuceneDescription());
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -585,7 +547,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
                     assertNotNull(result.getLuceneDescription());
                     assertThat(result.getTime(), greaterThan(0L));
                     assertNotNull(result.getTimeBreakdown());
-                    assertQueryProfileResult(result);
                 }
 
                 CollectorResult result = searchProfiles.getCollectorResult();
@@ -616,37 +577,6 @@ public class QueryProfilerIT extends ParameterizedOpenSearchIntegTestCase {
 
         SearchResponse resp = client().prepareSearch().setQuery(q).setProfile(false).get();
         assertThat("Profile response element should be an empty map", resp.getProfileResults().size(), equalTo(0));
-    }
-
-    private void assertQueryProfileResult(ProfileResult result) {
-        Map<String, Long> breakdown = result.getTimeBreakdown();
-        Long maxSliceTime = result.getMaxSliceTime();
-        Long minSliceTime = result.getMinSliceTime();
-        Long avgSliceTime = result.getAvgSliceTime();
-        if (concurrentSearchEnabled) {
-            assertNotNull(maxSliceTime);
-            assertNotNull(minSliceTime);
-            assertNotNull(avgSliceTime);
-            assertThat(breakdown.size(), equalTo(66));
-            for (QueryTimingType queryTimingType : QueryTimingType.values()) {
-                if (queryTimingType != QueryTimingType.CREATE_WEIGHT) {
-                    String maxTimingType = MAX_PREFIX + queryTimingType;
-                    String minTimingType = MIN_PREFIX + queryTimingType;
-                    String avgTimingType = AVG_PREFIX + queryTimingType;
-                    assertNotNull(breakdown.get(maxTimingType));
-                    assertNotNull(breakdown.get(minTimingType));
-                    assertNotNull(breakdown.get(avgTimingType));
-                    assertNotNull(breakdown.get(maxTimingType + TIMING_TYPE_COUNT_SUFFIX));
-                    assertNotNull(breakdown.get(minTimingType + TIMING_TYPE_COUNT_SUFFIX));
-                    assertNotNull(breakdown.get(avgTimingType + TIMING_TYPE_COUNT_SUFFIX));
-                }
-            }
-        } else {
-            assertThat(maxSliceTime, is(nullValue()));
-            assertThat(minSliceTime, is(nullValue()));
-            assertThat(avgSliceTime, is(nullValue()));
-            assertThat(breakdown.size(), equalTo(27));
-        }
     }
 
 }

